crates/nu-command/src/filters/split_by.rs

```diff
@@ -202,24 +202,18 @@ pub fn data_split(
                     }
                 }
                 _ => {
-                    return Err(ShellError::GenericError(
-                        "unsupported input".into(),
-                        "requires a table with one row for splitting".into(),
-                        Some(span),
-                        None,
-                        Vec::new(),
-                    ))
+                    return Err(ShellError::TypeMismatch {
+                        err_message: "requires a record to split".into(),
+                        span,
+                    })
```

Comment:
Coming from https://github.com/nushell/nushell/issues/10698 maybe `ShellError::OnlySupportsThisInputType` is the better choice here as it also reports the observed wrong type.

Reply:
I’ll update the rust error type

Reply:
Now using `OnlySupportsThisErrorType` and `PipelineEmpty` error types in 695a578a07df382fcbc0c72a89505dd45c4974a4

Reply:
Thanks! This looks good